pkg/ledger/ledger_test.go

```diff
@@ -437,11 +437,11 @@ func TestTransactionMetadata(t *testing.T) {
 		require.NoError(t, err)
 
 		meta, ok := tx.Metadata["a random metadata"]
-		assert.True(t, ok)
+		require.True(t, ok)
```

Comment:
I think that using `assert` here was more appropriate, as this is what we want to test. I like to separate error checking with `require` for things that are useful for the test but isn't what we are testing, and `assert` for things that we are actually testing. What do you think?

Reply:
I prefer require in all cases :D
I don't care about continue to test if the check fail.
It also complicate the testing process as we need to think about the legitimacy of the assert vs the require.
It is a mental burden for a minimal gain.
But i can change my mind.